### PR TITLE
Feature/add update board list multi

### PR DESCRIPTION
**Pull request by NicFisher**
### Overview

If a board list is created or updated, it will need to update the positions around it. This is ~~probably~~  **definitely** not the best solution.

### CODE REVIEW

File: lib/todo/boards/board_lists/board_list_manager.ex [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  alias Todo.Boards.BoardList
+  alias Todo.Boards.BoardLists.BoardListPositions
+  alias Todo.Repo
+
+  @moduledoc """
+  This module handles creating and updating board lists. When a board list is created or updated,
+  the positions of the other board lists on that board will be re-ordered if required.
+  """
+
+  # @spec create(%{}, %Board{}) :: {:ok %BoardList{}} | {:error, String.t}
```
Comment (NicFisher) [inline lib/todo/boards/board_lists/board_list_manager.ex:12]: I'm not sure why but I'm getting the this error for this spec
`(SyntaxError) syntax error before: '%'`

Comment (edgurgel) [reply lib/todo/boards/board_lists/board_list_manager.ex:12]: `{:ok %BoardList{}} `

Missing a comma here between `ok` and `%`

File: lib/todo/boards/board_lists/board_list_manager.ex [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+         updated_position <- String.to_integer(updated_position),
+         {:ok, %{board_list: board_list}} <-
+           update_board_list_and_positions(
+             board_list_changeset,
+             current_position,
+             updated_position,
+             board_id
+           ) do
+      {:ok, board_list}
+    else
```
Comment (edgurgel) [inline lib/todo/boards/board_lists/board_list_manager.ex:43]: I think you need an extra `{error` case here. When you use Repo.transaction with a multi the error tuple is like this:

https://hexdocs.pm/ecto/Ecto.Repo.html#c:transaction/2

`{:error, Ecto.Multi.name(), any(), %{required(Ecto.Multi.name()) => any()}}`

**Comment by NicFisher** [comment]
I tried leaving the `board_list_position_and_board_id_unique_index` and calling insert/update first in the multi, but it gives me this error:

```
{:error, :board_list,
 #Ecto.Changeset<
   action: :update,
   changes: %{name: "List 101", position: 2},
   errors: [
     unique_position: {"has already been taken",
      [
        constraint: :unique,
        constraint_name: "board_list_position_and_board_id_unique_index"
      ]}
   ],
   data: #Todo.Boards.BoardList<>,
   valid?: false
 >, %{}}
```

**Pull request merged**